agents/unix/conf/base/conf_sensor.c

```diff
@@ -0,0 +1,576 @@
+        {
+            te_string_append(&result, "%s%u", id == 0 ? "" : " ", id);
+        }
+        if (rc != 0)
+        {
+            te_string_free(&result);
+            sensors_free_chip_name(&match);
+            return rc;
+        }
+        id++;
+    }
+
+    *list = result.ptr;
+    sensors_free_chip_name(&match);
+    return 0;
+}
+
+static te_errno
+sensor_get_subfeature(const char *sensor_str, const char *data_id_str,
+                      const sensors_subfeature_type *subfeature_types,
+                      size_t sf_types_len, const char *function_name,
+                      char *value)
+{
+    int chip_nr = 0;
+    int feature_nr;
+    int err;
+    int i;
+    double subfeature_value;
+    sensors_chip_name match;
+    const sensors_chip_name *chip;
+    const sensors_feature *feature;
+    const sensors_subfeature *subfeature;
+
+    err = sensors_parse_chip_name(sensor_str, &match);
+    if (err)
+    {
+        ERROR("%s: sensors_parse_chip_name failed for chip name '%s': %s",
+              function_name, sensor_str, sensors_strerror(err));
+        return 0;
+    }
+
+    chip = sensors_get_detected_chips(&match, &chip_nr);
+    if (chip == NULL)
+    {
+        snprintf(value, RCF_MAX_VAL, "%s", "");
+        sensors_free_chip_name(&match);
+        return 0;
+    }
+
+    feature_nr = extract_last_int(data_id_str);
+    if (feature_nr == -1)
+    {
+        ERROR("%s: invalid data_id_str: %s", function_name, data_id_str);
+        sensors_free_chip_name(&match);
+        return 0;
+    }
+    feature = sensors_get_features(chip, &feature_nr);
+
+    for (i = 0; i < sf_types_len; i++)
+    {
+        subfeature = sensors_get_subfeature(chip, feature,
+                                            subfeature_types[i]);
+        if (subfeature != NULL)
+            break;
+    }
+    if (subfeature != NULL)
+    {
+        err = sensors_get_value(chip, subfeature->number,
+                                &subfeature_value);
+        if (err)
+        {
+            ERROR("%s: sensors_get_value failed: %s", function_name,
+                  sensors_strerror(err));
+            sensors_free_chip_name(&match);
+            return 0;
+        }
+        snprintf(value, RCF_MAX_VAL, "%f", subfeature_value);
+    }
+    else
+    {
+        snprintf(value, RCF_MAX_VAL, "%s", "");
+    }
+
+    sensors_free_chip_name(&match);
+    return 0;
+}
+
+/*
+ * Extracts a non-negative integral value from the end of string s.
+ * Returns:
+ *   on success, the int value scanned;
+ *   -1, if the string is NULL, empty, or not terminated by a number
+ */
+static int
+extract_last_int(const char* s)
+{
+    int value = -1;
+    const char *scanstr;
+
+    if (s && *s)
+    {
+        scanstr = s + strlen(s) - 1;
+        if (!isdigit(*scanstr))
+        {
+            return -1;
+        }
+        while (scanstr > s && isdigit(*(scanstr-1)))
+        {
+            scanstr--;
+        }
+        sscanf(scanstr, "%d", &value);
```

Comment:
You can actually get rid of most of this code: `sscanf(scanstr, "%*[^0-9]s%d", &value)` does exactly the same. But, of course, you'd better check that `sscanf` returns `1`.

Reply:
Sorry, not sure I understand what you mean; did you mean `sscanf(scanstr, "%*[^0-9]%d", &value)`?
This also does not work because we need the last number (at the end of string), not the first, e.g. for "in0_1" the function would return 1